thrall/app/lib/ThrallMessageConsumer.scala

```diff
@@ -83,9 +94,62 @@ class ThrallMessageConsumer(config: ThrallConfig, es: ElasticSearch, thrallMetri
   def deleteAllUsages(usage: JsValue) =
     Future.sequence( withImageId(usage)(id => es.deleteAllImageUsages(id)) )
 
-  def updateRcsRights(rights: JsValue) =
-    Future.sequence( withImageId(rights)(id => es.updateImageSyndicationRights(id, rights \ "data")) )
+  def refreshInferredSyndicationRights(imageJson: JsValue) = {
+    Future.sequence(
+      withImageId(imageJson) { id =>
+        val image = imageJson.as[Image]
+        GridLogger.info(s"upserting inferred rights", id)
+        es.updateImageSyndicationRights(id, image.syndicationRights)
+      }
+    )
+  }
+
+  def upsertSyndicationRights(rights: JsValue) = {
+    (rights \ "data").validate[SyndicationRights].fold(
+      err => {
+        val msg = s"Unable to parse message as SyndicationRights ${JsError.toJson(err).toString}"
+        GridLogger.error(msg)
+        Future.failed(SNSBodyParseError(msg))
+      },
+      rightsFromRcs => withImageId(rights) { id =>
+        es.getImage(id) map {
+          case Some(image) => {
+            GridLogger.info(s"Upserting syndication rights from RCS feed", image.id)
+            es.updateImageSyndicationRights(id, Some(rightsFromRcs))
+
+            image.userMetadata.map{edits => {
+              edits.photoshoot match {
+                case Some(photoshoot) if !rightsFromRcs.isInferred => {
+                  // Image is in a photoshoot and has real rights
+                  // Update the other images in the photoshoot
+                  es.getImagesWithInferredSyndicationRights(photoshoot, image)
+                    .map(initiateInferredRightsRefresh(_, rightsFromRcs))
+                  image
+                }
+                case _ => image
+              }
+            }}
+          }
+          case _ => {
+            GridLogger.info(s"image $id not found")
+            None
+          }
+        }
+      }
+    )
+  }
+
+  private def initiateInferredRightsRefresh(images: List[Image], rightsFromRcs: SyndicationRights) = {
+    GridLogger.info(s"initiating refresh of inferred rights for ${images.size} images")
+    val inferredRights = rightsFromRcs.copy(published = None)
+
+    images.foreach(image => {
+      val updatedImage = image.copy(syndicationRights = Some(inferredRights))
+      thrallNotifications.publish(Json.toJson(updatedImage), subject = "refresh-inferred-rights")
```

Comment:
This sends the entire image while we only need a small subset of the fields. Can we send only the id and the rights instead? (like we do for `upsert-rcs-rights`)

Reply:
Yes! Means we can reuse more code! PR updated now.